pkg/controllers/deprovisioning/consolidation.go

```diff
@@ -58,106 +71,146 @@ func (c *Consolidation) ShouldDeprovision(_ context.Context, n *state.Node, prov
 	return provisioner != nil && provisioner.Spec.Consolidation != nil && ptr.BoolValue(provisioner.Spec.Consolidation.Enabled)
 }
 
-// sortCandidates orders deprovisionable nodes by the disruptionCost
-func (c *Consolidation) SortCandidates(nodes []CandidateNode) []CandidateNode {
-	sort.Slice(nodes, func(i int, j int) bool {
-		return nodes[i].disruptionCost < nodes[j].disruptionCost
-	})
-	return nodes
-}
-
 // computeCommand generates a deprovisioning command given deprovisionable nodes
-func (c *Consolidation) ComputeCommand(ctx context.Context, attempt int, candidates ...CandidateNode) (Command, error) {
+func (c *Consolidation) ComputeCommand(ctx context.Context, candidates ...CandidateNode) (Command, error) {
 	// the last cluster consolidation wasn't able to improve things and nothing has changed regarding
 	// the cluster that makes us think we would be successful now
 	if c.lastConsolidationState == c.cluster.ClusterConsolidationState() {
 		return Command{action: actionDoNothing}, nil
 	}
+	// First sort the candidate nodes by disruption cost.
+	sort.Slice(candidates, func(i int, j int) bool {
+		return candidates[i].disruptionCost < candidates[j].disruptionCost
+	})
+
+	// We use this timestamp for the Consolidation TTL so that our TTL logic waits one time per consolidation loop.
+	validationTimestamp := time.Time{}
```

Comment:
nit: Do you need to set this and var this since it's going to get set anyways on L92?